calblueprint/meepanyar
# Abstracting Airtable Requests

## 🖋 PR Description

This PR creates abstractions of API requests to Airtable within requests.ts and also defines the schema and interface for our data based on the database.

### 🐞 Bug Report

- There were lots of bugs related to converting everything to the correct Record types, but I figured them out by searching through past codebases and referencing their methods.

### 🧪 Testing

How do we test your feature?

1. First you need to set up your .env file containing all the environment variables that you need to make requests to Airtable. Create a file in your directory called .env, and copy the contents of .env.example (that should be in this PR) into that file. REACT_APP_AIRTABLE_BASE_ID should be set equal to the ID of our base, which you can find here: https://airtable.com/appy54OSlg5bKS8rU/api/docs. REACT_APP_AIRTABLE_API_KEY should be set equal to your Airtable API key, which you can set up by going to API under your Account overview/settings on Airtable. BE SURE THAT YOU DO NOT COMMIT THIS FILE!! .gitignore should already prevent this from happening, but just verify that you don't accidentally commit it.

2. Once that's done, you can follow the example below to see how you can pull data from Airtable!
![backendpullexample](https://user-images.githubusercontent.com/35015640/97790110-0eecd580-1b83-11eb-94da-e9dbbe8f7d98.PNG)

3. You should be able to use npm start to run the code and see your results!

cc: @jenmhoang
